api/v1alpha1/labels.go

```diff
@@ -8,6 +8,8 @@ const (
 	CredentialTypeLabelValueGit   = "git"
 	CredentialTypeLabelValueHelm  = "helm"
 	CredentialTypeLabelValueImage = "image"
+	// TODO: Should we explicitly keep this label or agree that absence of this label should implicitly say that credential is generic?
+	CredentialTypeLabelValueGeneric = "generic"
```

Comment:
I think the advantage of this label is that we do not automatically make any Secret managed by Kargo, but instead, continue to allow them to be "opt-in".

Reply:
I was waffling on what to name this, but `generic` seems appropriate as it is the same term used by kubectl, i.e.

`kubectl create secret generic my-secret`

Reply:
Alright. I will keep this explicit. Thanks!